lib/cmap/connection_pool.js

```diff
@@ -449,13 +453,17 @@ function processWaitQueue(pool) {
     return;
   }
 
-  while (pool[kWaitQueue].length && pool.availableConnectionCount) {
+  while (pool[kWaitQueue].length) {
```

Comment:
Could we reuse `pool.waitQueueSize` here?

Reply:
sure! changed